articles/how-to-build-a-music-player-using-django/index.md

```diff
@@ -0,0 +1,364 @@
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'App'
+]
+```
+
+Now let's create our static and templates folders in our APP directory. The template folder will house our HTML files, while the static folder will house our CSS, Javascript, and other static files. To do this, run the command below on your command-line
+
+```bash
+# changing directory to our app
+$ cd App
+
+# creating  the static and templates folders
+$ mkdir templates
+$ mkdir static
+```
+
+In the templates folder, create a new file "index.html" and add the HTML code below:
+
+```HTML
+{% load static %}
+
+<!DOCTYPE html>
+<html lang="en">
+ <head>
+  <meta charset="utf-8"/>
+  <title>
+   My Music Player
+  </title>
+  <link href="https://netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet"/>
+  <link href="https://cdnjs.cloudflare.com/ajax/libs/mediaelement/4.2.7/mediaelementplayer.min.css" rel="stylesheet"/>
+  <link href="{% static './style.css' %}" rel="stylesheet"/>
+ </head>
+ <body>
+  <!-- partial:index.partial.html -->
+  <html>
+   <head>
+    <meta charset="utf-8"/>
+    <title>
+     Flat music player
+    </title>
+   </head>
+   <body>
+    <div class="contain">
+     <div class="container">
+      <div class="music-player">
+        {% for item in page_obj %}
+       <div class="cover">
+        <img alt="" src="{{item.image.url}}"/>
+       </div>
+       <div class="titre">
+        <h3>
+         {{item.artist}}
+        </h3>
+        <h1>
+         {{item.title}}
+        </h1>
+       </div>
+       <center><a href="{% if page_obj.has_previous %}?page={{ page_obj.previous_page_number }}{% endif %}"><i class="fa fa-step-backward fa-2x"></i></a> &nbsp; &nbsp; &nbsp; <a href="{% if page_obj.has_next %}?page={{ page_obj.next_page_number }} {% endif %}"><i class="fa fa-step-forward fa-2x"></i></a></center>
+       <div class="lecteur">
+        <audio class="fc-media" style="width: 100%;">
+         <source src="{% if item.audio_file %}{{item.audio_file.url}} {% else %} {{item.audio_link}} {% endif %}" type="audio/mp3"/>
+        </audio>
+
+       </div>
+       {% endfor %}
+      </div>
+     </div>
+    </div>
+   </body>
+  </html>
+  <!-- partial -->
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js">
+  </script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/mediaelement/4.2.7/mediaelement-and-player.min.js">
+  </script>
+  <script src="{% static './script.js' %}">
+  </script>
+ </body>
+</html>
+```
+
+In the code above we created the HTML attributes for our music player and made use of jinja; a python web templates engine to render our query set object in our **"page_obj"** context which we defined in our *views.py* file. Also the next and previous links, we rendered the pagination attributes which will separate our songs into a page per song format.
```

Comment:
At this point in the article, we aren't setting any **"page_obj"** in the *views.py* file before this step. Please check.